pkg/client/unversioned/clientcmd/loader.go

```diff
@@ -387,6 +393,25 @@ func WriteToFile(config clientcmdapi.Config, filename string) error {
 	return nil
 }
 
+func lockFile(filename string) error {
+	// TODO: find a way to do this with actual file locks. Will
+	// probably need seperate solution for windows and linux.
+	f, err := os.OpenFile(lockName(filename), os.O_CREATE|os.O_EXCL, 0)
```

Comment:
@dims I would definitely consider that for a later release, but I didn't want to try to rush something that hairy.